components/org.wso2.carbon.identity.api.server.claim.management/org.wso2.carbon.identity.rest.api.server.claim.management.v1/src/main/java/org/wso2/carbon/identity/rest/api/server/claim/management/v1/core/ServerClaimManagementService.java

```diff
@@ -1011,6 +1011,19 @@ private LocalClaimResDTO getLocalClaimResDTO(LocalClaim localClaim) {
             }
         }
 
+        String sharedProfileValueResolvingMethod =
+                claimProperties.remove(ClaimConstants.SHARED_PROFILE_VALUE_RESOLVING_METHOD);
+        if (StringUtils.isNotBlank(sharedProfileValueResolvingMethod)) {
+            try {
+                localClaimResDTO.setSharedProfileValueResolvingMethod(
+                        LocalClaimResDTO.SharedProfileValueResolvingMethodEnum.valueOf(
+                                sharedProfileValueResolvingMethod));
+            } catch (IllegalArgumentException e) {
+                // If the value is not a valid enum value, treat it as null.
+                localClaimResDTO.setSharedProfileValueResolvingMethod(null);
```

Comment:
Dont we need to fall back to the default method?

Reply:
The API layer can't decide the default method because system claim's behavior need to be decided based on system default ones, and other custom claim's default behavior will be FromOrigin.
This logic will be handled by BE OSGi service.
Therefore, API layer doesn't want to worry.